pkg/kubelet/stats/cri_stats_provider_windows.go

```diff
@@ -79,6 +81,73 @@ func (p *criStatsProvider) listContainerNetworkStats() (map[string]*statsapi.Net
 	return networkStats, nil
 }
 
+func (p *criStatsProvider) addCRIPodContainerStats(criSandboxStat *runtimeapi.PodSandboxStats,
+	ps *statsapi.PodStats, fsIDtoInfo map[runtimeapi.FilesystemIdentifier]*cadvisorapiv2.FsInfo,
+	containerMap map[string]*runtimeapi.Container,
+	podSandbox *runtimeapi.PodSandbox,
+	rootFsInfo *cadvisorapiv2.FsInfo,
+	updateCPUNanoCoreUsage bool) {
+	for _, criContainerStat := range criSandboxStat.Windows.Containers {
+		container, found := containerMap[criContainerStat.Attributes.Id]
+		if !found {
+			continue
+		}
+		// Fill available stats for full set of required pod stats
+		cs := p.makeWinContainerStats(criContainerStat, container, rootFsInfo, fsIDtoInfo, podSandbox.GetMetadata())
+		ps.Containers = append(ps.Containers, *cs)
+	}
+}
+
+func (p *criStatsProvider) makeWinContainerStats(
+	stats *runtimeapi.WindowsContainerStats,
+	container *runtimeapi.Container,
+	rootFsInfo *cadvisorapiv2.FsInfo,
+	fsIDtoInfo map[runtimeapi.FilesystemIdentifier]*cadvisorapiv2.FsInfo,
+	meta *runtimeapi.PodSandboxMetadata) *statsapi.ContainerStats {
+	result := &statsapi.ContainerStats{
+		Name: stats.Attributes.Metadata.Name,
+		// The StartTime in the summary API is the container creation time.
+		StartTime: metav1.NewTime(time.Unix(0, container.CreatedAt)),
+		CPU:       &statsapi.CPUStats{},
+		Memory:    &statsapi.MemoryStats{},
+		Rootfs:    &statsapi.FsStats{},
+		// UserDefinedMetrics is not supported by CRI.
+	}
+	if stats.Cpu != nil {
+		criCPU := stats.Cpu
+		result.CPU = &statsapi.CPUStats{
+			Time:                 metav1.NewTime(time.Unix(0, criCPU.Timestamp)),
+			UsageNanoCores:       valueOfUInt64Value(criCPU.UsageNanoCores),
+			UsageCoreNanoSeconds: valueOfUInt64Value(criCPU.GetUsageCoreNanoSeconds()),
+		}
+	}
+	if stats.Memory != nil {
+		criMemory := stats.Memory
+		result.Memory = &statsapi.MemoryStats{
+			Time:            metav1.NewTime(time.Unix(0, criMemory.Timestamp)),
+			AvailableBytes:  valueOfUInt64Value(criMemory.AvailableBytes),
+			WorkingSetBytes: valueOfUInt64Value(criMemory.WorkingSetBytes),
+			PageFaults:      valueOfUInt64Value(criMemory.PageFaults),
+		}
+	}
+	if stats.WritableLayer != nil {
+		criRootfs := stats.WritableLayer
+		result.Rootfs = &statsapi.FsStats{
+			Time:      metav1.NewTime(time.Unix(0, criRootfs.Timestamp)),
+			UsedBytes: valueOfUInt64Value(criRootfs.UsedBytes),
```

Comment:
We only have UsedBytes? We don't have CapacityBytes?

Note: On the Linux implementation, it also gets information from ``imageFsInfo``. Don't we have to as well?